src/appWithTranslation.tsx

```diff
@@ -17,11 +17,11 @@ type AppProps = NextJsAppProps & {
 
 export let globalI18n: I18NextClient | null = null
 
-export const appWithTranslation = (
-  WrappedComponent: React.ComponentType<AppProps> | React.ElementType<AppProps>,
+export const appWithTranslation = <Props extends AppProps = AppProps>(
+  WrappedComponent: React.ComponentType<Props> | React.ElementType<AppProps>,
```

Comment:
```suggestion
  WrappedComponent: React.ComponentType<Props> | React.ElementType<Props>,
```

To answer your question about why we need `ElementType` – it's just for the sake of comprehensiveness. See the type def:

```ts
type ElementType<P = any> =
  {
      [K in keyof JSX.IntrinsicElements]: P extends JSX.IntrinsicElements[K] ? K : never
  }[keyof JSX.IntrinsicElements] |
  ComponentType<P>;
```

Reply:
The reason that I asked in the first place is because of the following error when providing ElementType with a generic type parameter like `Props` in our case
![Screenshot_10-11-2021_19:11:20](https://user-images.githubusercontent.com/36507050/141169534-4876cefe-20e1-49bc-b616-c75488b4570b.png)
![Screenshot_10-11-2021_19:09:49](https://user-images.githubusercontent.com/36507050/141169392-3c54d8ee-7d6b-4a73-915c-e2b7ba539d07.png)

Reply:
OK, I'm fine with just using `React.ComponentType<Props>`, but we'll need to wait for the v9 release to publish a change like that (should be this week).

If you're happy with that, please change the target of your PR to: https://github.com/isaachinman/next-i18next/pull/1512

Reply:
I don't have any problem with that